### PR TITLE
[JENKINS-38865] Move the AST classes from groovy to Java

### PR DESCRIPTION
**Pull request by stephenc**
Just creating a PR for CI verification... this PR will not be merged

### CODE REVIEW

**Comment by abayer** [comment]
Merged in latest master after merging #19

File: pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTPostStage.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public String toString() {
+        return "ModelASTPostStage{" +
+                "conditions=" + getConditions() +
+                "}";
+    }
+
+    @Override
+    public void validate(ModelValidator validator) {
+        validator.validateElement(this);
+        super.validate(validator);
```
Comment (rsandell) [inline pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTPostStage.java]: Potential double validation

Comment (stephenc) [inline pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTPostStage.java]: Nope... check the tests... this one will call the specific validation and then the generic validation

**Comment by rsandell** [comment]
:bee:

**Comment by abayer** [comment]
Yup, :bee: - merging!

**Pull request merged**